base/sort.jl

```diff
@@ -335,14 +335,13 @@ julia> searchsorted([1=>"one", 2=>"two", 2=>"two", 4=>"four"], 2=>"two", by=firs
 """
     searchsortedfirst(v, x; by=identity, lt=isless, rev=false)
 
-Return the index of the first value in `v` greater than or equivalent to `x`.
-If `x` is greater than all values in `v`, return `lastindex(v) + 1`.
+Return the index of the first value in `v` that is not ordered before `x`.
+If all values in `v` are ordered before `x`, return `lastindex(v) + 1`.
 
 The vector `v` must be sorted according to the order defined by the keywords.
-`insert!`ing `x` at the returned index will maintain the sorted order. Refer to
```

Comment:
This remark is pretty subtle; as stated, it does not hold for `searchsortedlast`.

If we consider stability, inserting at the index searchsortedfirst returns is equivalent to inserting at the beginning and sorting, while inserting at the index returned by searchsortedlast + 1 is equivalent to inserting at the end and then sorting.

I think it’s worth keeping because inserting while maintaining sorted order is a pretty common use case of searchsortedfirst, and it is not immediately obvious that searchsortedfirst is the right choice for that and searchsortedlast is not.

Reply:
Would the correct statement for `searchsortedlast` be `insert!ing x immediately after the returned index will maintain the sorted order.`?

Reply:
That is a true statement.